### ISSUE TITLE
`get_predicted.iv_robust()` issues

### ISSUE THREAD

**Issue by vincentarelbundock** [easystats/insight#548]
Two issues with `get_predicted.iv_robust()`:

1.  No default value for the `predict` argument
2.  Does not recognize the `data` argument when `ci` is not NULL. Notice that the two last commands produce outputs of different dimensions:

``` r
library(ivreg)
library(estimatr)
library(insight)
data(Kmenta, package = "ivreg")
model <- iv_robust(Q ~ P + D | D + F + A, 
                   se_type = "stata",
                   data = Kmenta)

get_predicted(model, data = head(Kmenta))
#> Error: Please supply a value for the `predict` argument.

get_predicted(model, data = head(Kmenta), predict = "expectation")
#> Predicted values:
#> 
#> [1]  97.64186  99.88472  99.80404 100.01418 102.10090 101.96632
#> 
#> NOTE: Confidence intervals, if available, are stored as attributes and can be accessed using `as.data.frame()` on this output.

get_predicted(model, data = head(Kmenta), predict = "expectation", ci = .95)
#> Predicted values:
#> 
#>  [1]  97.64186  99.88472  99.80404 100.01418 102.10090 101.96632 102.42197
#>  [8] 102.96591 101.47486 100.32800  95.54263  94.71564  96.13309  99.02829
#> [15] 103.83900 103.65546 103.83496 102.08027 103.63146 106.90043
#> 
#> NOTE: Confidence intervals, if available, are stored as attributes and can be accessed using `as.data.frame()` on this output.
```

**Comment by strengejacke** [comment]
It fails due to the missing model matrix:

``` r
library(ivreg)
library(estimatr)
library(insight)
data(Kmenta, package = "ivreg")
model <- iv_robust(Q ~ P + D | D + F + A, 
                   se_type = "stata",
                   data = Kmenta)

model.matrix(model)
#> Error in eval(predvars, data, env): object 'Q' not found
```

<sup>Created on 2022-03-23 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by strengejacke** [comment]
What would be a appropriate matrix here? Like this? 

``` r
library(ivreg)
library(estimatr)
library(insight)
data(Kmenta, package = "ivreg")
model <- iv_robust(Q ~ P + D | D + F + A, 
                   se_type = "stata",
                   data = Kmenta)

model.matrix(find_formula(model)[[1]], data = head(Kmenta))
#>      (Intercept)       P     D
#> 1922           1 100.323  87.4
#> 1923           1 104.264  97.6
#> 1924           1 103.435  96.7
#> 1925           1 104.506  98.2
#> 1926           1  98.001  99.8
#> 1927           1  99.456 100.5
#> attr(,"assign")
#> [1] 0 1 2
```

<sup>Created on 2022-03-23 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by strengejacke** [comment]
`iv_robust` has no predict method. I think this is rather something which needs to be addressed in _estimatr_ first?

**Comment by strengejacke** [comment]
Nevermind, I think I solved the issue:

``` r
library(ivreg)
library(estimatr)
library(insight)
data(Kmenta, package = "ivreg")
model <- iv_robust(Q ~ P + D | D + F + A, 
                   se_type = "stata",
                   data = Kmenta)

get_predicted(model, data = head(Kmenta))
#> Predicted values:
#> 
#> [1]  97.64186  99.88472  99.80404 100.01418 102.10090 101.96632
#> 
#> NOTE: Confidence intervals, if available, are stored as attributes and can be accessed using `as.data.frame()` on this output.

get_predicted(model, data = head(Kmenta), predict = "expectation")
#> Predicted values:
#> 
#> [1]  97.64186  99.88472  99.80404 100.01418 102.10090 101.96632
#> 
#> NOTE: Confidence intervals, if available, are stored as attributes and can be accessed using `as.data.frame()` on this output.

get_predicted(model, data = head(Kmenta), predict = "expectation", ci = .95)
#> Predicted values:
#> 
#> [1]  97.64186  99.88472  99.80404 100.01418 102.10090 101.96632
#> 
#> NOTE: Confidence intervals, if available, are stored as attributes and can be accessed using `as.data.frame()` on this output.

get_predicted(model, data = head(Kmenta), predict = "expectation", ci = .95) |> as.data.frame()
#>   Predicted        SE    CI_low  CI_high
#> 1  97.64186 0.5835286  96.41073  98.8730
#> 2  99.88472 0.5023223  98.82492 100.9445
#> 3  99.80404 0.4832473  98.78448 100.8236
#> 4 100.01418 0.5032670  98.95238 101.0760
#> 5 102.10090 0.5557442 100.92838 103.2734
#> 6 101.96632 0.5089377 100.89256 103.0401


m <- ivreg(Q ~ P + D | D + F + A, data = Kmenta)

get_predicted(m, data = head(Kmenta))
#> Predicted values:
#> 
#> [1]  97.64186  99.88472  99.80404 100.01418 102.10090 101.96632
#> 
#> NOTE: Confidence intervals, if available, are stored as attributes and can be accessed using `as.data.frame()` on this output.

get_predicted(m, data = head(Kmenta), predict = "expectation")
#> Predicted values:
#> 
#> [1]  97.64186  99.88472  99.80404 100.01418 102.10090 101.96632
#> 
#> NOTE: Confidence intervals, if available, are stored as attributes and can be accessed using `as.data.frame()` on this output.

get_predicted(m, data = head(Kmenta), predict = "expectation", ci = .95)
#> Predicted values:
#> 
#> [1]  97.64186  99.88472  99.80404 100.01418 102.10090 101.96632
#> 
#> NOTE: Confidence intervals, if available, are stored as attributes and can be accessed using `as.data.frame()` on this output.

get_predicted(m, data = head(Kmenta), predict = "expectation", ci = .95) |> as.data.frame()
#>   Predicted        SE    CI_low   CI_high
#> 1  97.64186 0.6609226  96.24744  99.03629
#> 2  99.88472 0.5996838  98.61950 101.14995
#> 3  99.80404 0.5644068  98.61325 100.99483
#> 4 100.01418 0.6049375  98.73787 101.29049
#> 5 102.10090 0.5164204 101.01135 103.19045
#> 6 101.96632 0.4737788 100.96673 102.96591
```

<sup>Created on 2022-03-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by strengejacke** [comment]
closed in #549
